src/components/recently-viewed.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import { useHover } from 'use-events';
+import { Flex, Box, Text, FlexProps, BoxProps } from '@blockstack/ui';
+
+import { Tag } from '@components/tags';
+import { truncateMiddle } from '@common/utils';
+import { Transaction } from '@models/transaction.interface';
+import Link from 'next/link';
+
+dayjs.extend(relativeTime);
+
+interface RecentlyViewedProps extends FlexProps {
+  transactions: Transaction[];
+}
+
+export const RecentlyViewed = ({ transactions, ...rest }: RecentlyViewedProps) => {
+  if (!transactions.length) return null;
+  return (
+    <Flex
+      flexDirection="column"
+      border="1px solid #E1E3E8"
+      width={['100%', '100%', '544px']}
+      borderRadius="6px"
+      maxHeight="308px"
+      style={{ overflowY: 'auto' }}
+      {...rest}
+    >
+      <Box mx="base" mt={['base-tight', 'base-loose']} mb="tight">
+        <Text color="#677282" fontWeight={600} fontSize="11px" lineHeight="base" style={{ textTransform: 'uppercase' }}>
+          Recently Viewed
+        </Text>
+      </Box>
+      <RecentlyViewedList transactions={transactions} />
+    </Flex>
+  );
+};
+
+interface RecentlyViewedListProps extends BoxProps {
+  transactions: Transaction[];
+}
+
+export const RecentlyViewedList = ({ transactions, ...rest }: RecentlyViewedListProps) => {
+  return (
+    <Box width="100%" {...rest}>
+      {transactions.map((tx, i) => (
+        <RecentlyViewedListItem transaction={tx} isLast={transactions.length - 1 === i} />
+      ))}
+    </Box>
+  );
+};
+
+interface RecentlyViewedListItemProps extends BoxProps {
+  transaction: Transaction;
+  isLast: boolean;
+}
+
+export const RecentlyViewedListItem = ({ transaction, isLast }: RecentlyViewedListItemProps) => {
+  const [isHovered, bind] = useHover();
+  return (
+    <Link href="/txid/[txid]" as={`/txid/${transaction.tx_id}`}>
```

Comment:
The flex should have `as="a"` and the Link should have `passHref`: https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag